### PR TITLE
Use interface to access API

### PR DESCRIPTION
**Pull request by peppy**
Allows for better testability.

### CODE REVIEW

**Review by smoogipoo — APPROVED** [review]

File: osu.Game/Online/API/DummyAPIAccess.cs [hunk 1]
```diff
@@ -11,21 +11,45 @@ public class DummyAPIAccess : IAPIProvider
...
         });
 
         public bool IsLoggedIn => true;
 
-        public void Update()
-        {
-        }
+        public string ProvidedUsername => LocalUser.Value.Username;
+
+        public string Endpoint => "https://test.com";
```
Comment (smoogipoo) [inline osu.Game/Online/API/DummyAPIAccess.cs]:
```suggestion
        public string Endpoint => "http://localhost";
```

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Online/API/DummyAPIAccess.cs [hunk 2]
```diff
@@ -11,21 +11,45 @@ public class DummyAPIAccess : IAPIProvider
...
+
+        public APIState State => LocalUser.Value.Id == 1 ? APIState.Offline : APIState.Online;
```
Comment (smoogipoo) [inline osu.Game/Online/API/DummyAPIAccess.cs]: Huh? This doesn't work together with `Login()`. Why not just store a simple bool?

**Review by smoogipoo — DISMISSED** [review]

File: osu.Game/Online/API/DummyAPIAccess.cs [hunk 3]
```diff
@@ -11,21 +11,45 @@ public class DummyAPIAccess : IAPIProvider
         public Bindable<User> LocalUser { get; } = new Bindable<User>(new User
         {
             Username = @"Dummy",
-            Id = 1,
+            Id = 1001,
```
Comment (smoogipoo) [inline osu.Game/Online/API/DummyAPIAccess.cs]: Surely you don't want this the default, otherwise the state is going to be online by default?

Comment (peppy) [reply osu.Game/Online/API/DummyAPIAccess.cs]: It's what we expect in tests, yeah.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**